clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -639,19 +647,31 @@ private void maybeUpdate(long now, Node node) {
                 ClientRequest clientRequest = request(now, nodeConnectionId, metadataRequest);
                 log.debug("Sending metadata request {} to node {}", metadataRequest, node.id());
                 doSend(clientRequest, now);
-            } else if (connectionStates.canConnect(nodeConnectionId, now)) {
+                return requestTimeoutMs;
+            }
+
+            if (lastConnectAttemptNodeId != null &&
```

Comment:
As it's currently written, I'm not sure I see the point of this since we don't use `lastConnectAttemptNodeId` for sending.